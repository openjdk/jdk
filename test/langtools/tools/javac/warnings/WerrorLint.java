/*
 * @test /nodynamiccopyright/
 * @bug 8349847
 *
 * @compile                             -XDrawDiagnostics -Xlint:none                         WerrorLint.java
 * @compile                             -XDrawDiagnostics -Xlint:none     -Werror             WerrorLint.java
 * @compile                             -XDrawDiagnostics -Xlint:none     -Werror:empty       WerrorLint.java
 * @compile                             -XDrawDiagnostics -Xlint:none     -Werror:strictfp    WerrorLint.java
 * @compile/ref=WerrorLint.w2.out       -XDrawDiagnostics -Xlint:all                          WerrorLint.java
 * @compile/fail/ref=WerrorLint.e2.out  -XDrawDiagnostics -Xlint:all      -Werror             WerrorLint.java
 * @compile/fail/ref=WerrorLint.e2.out  -XDrawDiagnostics -Xlint:all      -Werror:empty       WerrorLint.java
 * @compile/fail/ref=WerrorLint.e2.out  -XDrawDiagnostics -Xlint:all      -Werror:strictfp    WerrorLint.java
 * @compile/ref=WerrorLint.w1.out       -XDrawDiagnostics                                     WerrorLint.java
 * @compile/fail/ref=WerrorLint.e1.out  -XDrawDiagnostics                 -Werror             WerrorLint.java
 * @compile/ref=WerrorLint.w1.out       -XDrawDiagnostics                 -Werror:empty       WerrorLint.java
 * @compile/fail/ref=WerrorLint.e1.out  -XDrawDiagnostics                 -Werror:strictfp    WerrorLint.java
 */

class WerrorLint {
    strictfp void m() {             // [strictfp] - this category is enabled by default
        if (hashCode() == 1) ;      // [empty]    - this category is disabled by default
    }
}
