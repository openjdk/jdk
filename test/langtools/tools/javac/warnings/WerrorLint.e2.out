WerrorLint.java:20:19: compiler.warn.strictfp
WerrorLint.java:21:30: compiler.warn.empty.if
- compiler.err.warnings.and.werror
1 error
2 warnings
